### PR TITLE
Df remove xprintf

### PR DESCRIPTION
**Pull request by dflanderka**
- remove INPUT_CHECK, xprintf, Semchem
- solve #647

### CODE REVIEW

File: src/system/system.cc [hunk 1]
```diff
@@ -202,7 +56,7 @@ void *xmalloc( size_t size )
 
 	if (size == 0 ) size++;
 	rc = malloc( size );
-	if ( rc == NULL ) xprintf(Err ,"Not enough memory for allocating %u bytes\n", size );
+	ASSERT_PTR( rc )(size).error("Not enough memory for allocating %u bytes\n" );
 
 	return(rc);
 }*/
```
Comment (jbrezmorf) [inline src/system/system.cc:62]: remove xmalloc

File: src/system/system.cc [hunk 2]
```diff
@@ -236,7 +90,7 @@ void * xrealloc( void * ptr, size_t size )
     void * rc;
 
     rc = realloc( ptr, size );
-    if ( rc == NULL ) xprintf(Err ,"Not enough memory for allocating %u bytes\n", size );
+    ASSERT_PTR( rc )(size).error("Not enough memory for allocating %u bytes\n" );
 
     return(rc);
 }*/
```
Comment (jbrezmorf) [inline src/system/system.cc:96]: remove xrealloc

**Review by jbrezmorf — COMMENTED** [review]
Navrhuji drobne upravy.

File: src/mesh/mesh.h [hunk 3]
```diff
@@ -80,10 +80,28 @@ class Mesh {
...
             << "Element id: " << EI_ElemLast::val << " on region name: " << EI_RegLast::val << "\n"
             << "Element id: " << EI_ElemNew::val << " on region name: " << EI_RegNew::val << "\n");
+    DECLARE_EXCEPTION(ExcElmWrongOrder,
+            << "Element IDs in non-increasing order, ID: " << EI_ElemId::val << "\n");
+    DECLARE_EXCEPTION(ExcRegionElmDiffDim,
+    		<< "User defined region " << EI_Region::qval << " (id " << EI_RegIdx::val
+            << ") by 'From_Elements' cannot have elements of different dimensions.\n"
+            << "Thrown due to: dim " << EI_Dim::val << " neq dim " << EI_DimOther::val << " (ele id " << EI_ElemId::val << ").\n"
+            << "Split elements by dim, create separate regions and then possibly use Union.\n" );
+    DECLARE_EXCEPTION(ExcTooMatchingIds,
```
[the remark this answers is not in the document]
Comment (dflanderka) [reply src/mesh/mesh.h:101]: The change is on the next line.

**Pull request merged**